### PR TITLE
Budget app project

### PR DESCRIPTION
**Pull request by AlyAtalla**
welcome to ROR Capstone Project: budget-app

Project requirements
Design
You should follow these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:
Colors.
Typography: font face, size and weight.
Layout: composition and space between elements.
Interactions
 Splash screen
A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
Sign up and log in pages

The user should be able to register in the app with full name, email and password (all mandatory).

The user can log into the app using email and password.

If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

 Home page (categories page)

When the user logs in, they are presented with the categories page.

For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.

When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.

There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

 Transactions page

For a given category, the list of transactions is presented, ordered by the most recent.

At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).

There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.

When the user clicks on the "Back" button (<), the user navigates to the home page.

 "Add a new category" page

The user fills out a form to create a new category, indicating their name and icon (both mandatory).

The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.

When the user clicks on the "Back" button (<), the user navigates to the home page.

 "Add a new transaction" page

The user fills out a form to create a new transaction with:

name (mandatory)

amount (mandatory)

categories (mandatory at least one)

The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.

When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

 Testing requirements
Create unit and integration tests for all the most important components of your RoR application.

specs

 Technical requirements

You should use Postgres as your database.

You should use devise for authentication.

You should validate all user input to make sure that anyone with bad intentions cannot compromise your app.

You can use a view template engine of your choice (.erb, .slim, .haml).

The project should be deployed and accessible online.

Your database schema should reflect the following structure:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,193 @@
...
+
+- [Live Demo Link](Comming soon)
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- Presentation Video -->
+## 🎥 Presentation Video <a name="presentation-video"></a>
+
+- [Presentation Video Link](comming soon)
+
```
Comment (DammyShittu) [inline README.md]: - [ ] Kindly deploy your app and provide its live demo link on the readme documentation.

- [ ] Kindly make a presentation of your app, preferably on Loom, demonstrating features of the app. The video should not be more than 5 minutes long and kindly also highlight code or feature you find interesting.

File: app/views/exchanges/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<nav class="navbar" style="background-color: #3778c2;" role="navigation">
+  <div class="navbar-brand">
+    <a href="/" class="navbar-item">
```
Comment (DammyShittu) [inline app/views/exchanges/new.html.erb:3]: - [ ] When a user is on the add new transaction form page, and they click the arrow "back" button (<) they should be taken to the transactions index page of the category, not the category page or root page.

  image

File: app/views/exchanges/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  <% @transactions.each do |transaction| %>
+    <li>
+      <p class = ""><%= transaction.name %></p>
+      <p class = "">$<%= transaction.amount %></p>
+    </li> 
+
+    <div class="button-container">
+    <%= button_to "Delete Transaction", category_exchange_path(@category, transaction), method: :delete, class: 'button is-light is-small delete_btn', style: 'box-shadow: 0 2px 4px #36363682; border-radius: 8px;margin-top:10px;' %> 
+ <%= link_to "Update Transaction", edit_category_exchange_path(@category, transaction), class: 'button is-light is-small delete_btn', style: 'box-shadow: 0 2px 4px #36363682; border-radius: 8px;' %>
+    </div>
```
Comment (DammyShittu) [inline app/views/exchanges/index.html.erb:37]: - [ ] The transactions on the transactions index page should be redesigned to match the design. You should indicate the time the item was created and include some kind of image to maintain the positioning of elements as requested in the project.

  Your Implementation | Behance Design
  :---:|:---:
  image | image

File: spec/integration/index_spec.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  describe 'Index page' do
+    before(:each) do
+      # Set up any necessary data or mocks here
+    end
+
+    before do
+      render
+    end
+  end
+end
```
Comment (DammyShittu) [inline spec/integration/index_spec.rb]: - [ ] You only have unit model tests. You have no integration tests. Kindly add enough integration tests to cover several combinations of user actions in your app.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @AlyAtalla,

## Highlights
✔️ Splash screen
✔️ Positioned new categories and transactions buttons
✔️ Ordered transactions by most recent

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by binhussen — APPROVED** [review]
Hi @AlyAtalla , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉
IApproveSimonCowellGIF (6)

## Highlights
- Sticked Given Design ✔️
- Well Stractured Code ✔️
- Used Git flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**